pkg/cephfs/volumemounter.go

```diff
@@ -113,3 +114,63 @@ func unmountVolume(mountPoint string) error {
 func createMountPoint(root string) error {
 	return os.MkdirAll(root, 0750)
 }
+
+func getAvailableMounters() []string {
+	var ms []string
+
+	fuseMounterProbe := exec.Command("ceph-fuse", "--version")
+	kernelMounterProbe := exec.Command("mount.ceph")
+
+	if fuseMounterProbe.Run() == nil {
+		ms = append(ms, volumeMounter_fuse)
+	}
+
+	if kernelMounterProbe.Run() == nil {
+		ms = append(ms, volumeMounter_kernel)
+	}
+
+	return ms
+}
+
+func newMounter(volOptions *volumeOptions) (volumeMounter, error) {
+	// Get the mounter from the configuration
+
+	wantMounter := volOptions.Mounter
+
+	if wantMounter == "" {
+		wantMounter = DefaultVolumeMounter
+	}
+
+	// Verify that it's available
+
+	availableMounters := getAvailableMounters()
```

Comment:
how about making `availableMounters` a global var? This will save a ceph-fuse call each time filesystem is to be mounted.

Reply:
Sure, I wasn't too sure about the use-cases here, TBH. Installing either of the mounters at runtime is not very plausible.